jni/src/test/java/com/horizen/merkletreenative/MerkleTreeTest.java

```diff
@@ -1,314 +1,314 @@
+//        InMemoryOptimizedMerkleTree mht = InMemoryOptimizedMerkleTree.init(6, numLeaves);
+//        int numLeaves = 64;
+//
+//        // Append leaves to the tree
+//        for (int i = 0; i < numLeaves/2; i ++) {
+//            FieldElement leaf = FieldElement.createRandom(i);
+//            testLeaves.add(leaf);
+//            mht.append(leaf);
+//        }
+//        for (int i = numLeaves/2; i < numLeaves; i ++) {
+//            FieldElement leaf = FieldElement.createFromLong(0L);
+//            testLeaves.add(leaf);
+//        }
+//
+//        //Finalize the tree and get the root
+//        mht.finalizeTreeInPlace();
+//        FieldElement mhtRoot = mht.root();
+//
+//        for (int i = 0; i < numLeaves; i ++) {
+//
+//            // Get/Verify Merkle Path
+//            MerklePath path = mht.getMerklePath((long)i);
+//            assertTrue("Merkle Path must be verified", path.verify(testLeaves.get(i), mhtRoot));
+//
+//            // Serialization/Deserialization test
+//            byte[] merklePathBytes = path.serialize();
+//            MerklePath pathDeserialized = MerklePath.deserialize(merklePathBytes);
+//            assertTrue("Deserialized Merkle Path must be verified", pathDeserialized.verify(testLeaves.get(i), mhtRoot));
+//
+//            if (i == 0) { // leftmost check
+//                assertTrue("Path must be the leftmost", path.isLeftmost());
+//            }
+//            else if (i == (numLeaves / 2) - 1) { // non-empty rightmost check
+//                assertTrue("Path must be the non-empty rightmost", path.isNonEmptyRightmost());
+//            }
+//            else if (i == numLeaves - 1) { //rightmost check
+//                assertTrue("Path must be the rightmost", path.isRightmost());
+//            }
+//            else { // Other cases check
+//                assertFalse("Path must not be the leftmost", path.isLeftmost());
+//                assertFalse("Path must not be the rightmost", path.isRightmost());
+//
+//                if (i < (numLeaves / 2) - 1) {
+//                    assertFalse("Path must not be the non-empty rightmost", path.isNonEmptyRightmost());
+//                }
+//            }
+//
+//            assertEquals("Leaf index computed from path must be correct", i, path.leafIndex());
+//
+//            // apply() test
+//            FieldElement rootComputed = path.apply(testLeaves.get(i));
+//            assertEquals("Root computed out of Merkle Path must be the same", rootComputed, mhtRoot);
+//            rootComputed.freeFieldElement();
+//
+//            // Free paths
+//            path.freeMerklePath();
+//            pathDeserialized.freeMerklePath();
+//        }
+//
+//        // Free memory
+//        mht.freeInMemoryOptimizedMerkleTree();
+//        mhtRoot.freeFieldElement();
+//        for (FieldElement leaf: testLeaves)
+//            leaf.freeFieldElement();
+//
+//    }
+//
+//    @Test
+//    public void testNonEmptyRightmost() {
```

Comment:
Don't see any usage of hardcoded data here. Maybe we can keep it ?

Reply:
This test uses expectedRootBytes hardcoded array